File: bindings/imgui_bundle/demos/demos_immvision/demo_immvision_inspector.py
import os.path

from imgui_bundle import immvision, immapp, imgui
import cv2


def fill_inspector():
    this_dir = os.path.dirname(__file__)
    image_files = ["dmla.jpg", "house.jpg", "tennis.jpg", "world.jpg"]
    for image_file in image_files:
        img = cv2.imread(f"{this_dir}/../assets/images/{image_file}")
        immvision.inspector_add_image(img, legend=image_file)


@immapp.static(inited=False)
def gui():
    if not gui.inited:
        fill_inspector()
        gui.inited = True

    immvision.inspector_show()


def main():
    immapp.run(gui, window_size=(1000, 800))


if __name__ == "__main__":
    main()

File: demos_cpp/demos_immvision/demo_immvision_inspector.cpp
#include <opencv2/imgcodecs.hpp>
#include "immvision/immvision.h"
#include "immapp/immapp.h"
#include "hello_imgui/hello_imgui.h"
#include <vector>
#include <string>


void FillInspector()
{
    std::vector<std::string> imagefiles = { "dmla.jpg", "house.jpg", "tennis.jpg", "world.jpg"};
    for (auto imageFile: imagefiles)
    {
        cv::Mat img = cv::imread(std::string("assets/images/") + imageFile);
        ImmVision::Inspector_AddImage(img, imageFile);
    }
}


void gui()
{
    static bool inited = false;
    if (!inited)
    {
        FillInspector();
        inited = true;
    }

    ImmVision::Inspector_Show();
}


int main()
{
    ImmApp::Run(gui, "inspector", false, false, {1000, 800});
}
